### PR TITLE
CAN rebase

### PR DESCRIPTION
**Pull request by andreaskuster**

### CODE REVIEW

File: lib/CAN/CANBus.cpp [hunk 1]
```diff
@@ -1,63 +1,152 @@
...
+  CAN.setPins(CAN_RX, CAN_TX);
+  CAN.onReceive(onReceiveForwarder); // not sure if works
 
-  CAN_cfg.speed = CAN_SPEED; // MPPT & BMS are both running on 125KBPS by default
-  CAN_cfg.tx_pin_id = CAN_TX;
-  CAN_cfg.rx_pin_id = CAN_RX;
-  CAN_cfg.rx_queue = xQueueCreate(CAN_RX_QUEUE, sizeof(CAN_frame_t));
-  ESP32Can.CANInit();
+  if (!CAN.begin(CAN_SPEED)) {
+    // opening CAN failed :,(
```
Comment (KarlSchu) [inline lib/CAN/CANBus.cpp:36]: Does it mean it doesn't work?

**Review by KarlSchu — COMMENTED** [review]
We used only cout streams and fmt::format(). This should be fixed next time.

File: lib/CAN/CANPacket.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+uint64_t CANPacket::getData_ui64() { return this->data; }
+
+// ToDo Test unsigned->signed on ESP32 runtime
+int64_t CANPacket::getData_i64() {
+  int64_t result = this->data & 0xFFFFFFFFFFFFFFFF;
+  return result;
+}
+
+uint32_t CANPacket::getData_ui32(uint8_t index) { return (uint32_t)(this->data >> (32 * index)) & 0x00000000FFFFFFFF; }
```
Comment (KarlSchu) [inline lib/CAN/CANPacket.cpp:43]: Why are uint... and int... handled in same manner? Could simplify and remove and type cast at usage?

File: lib/CAN/CANBus.h [hunk 3]
```diff
@@ -2,21 +2,26 @@
...
 
-#ifndef SOLAR_CAR_CONTROL_SYSTEM_CAN_H
-#define SOLAR_CAR_CONTROL_SYSTEM_CAN_H
-
-#include <CAN_config.h>
-#include <ESP32CAN.h>
+#include "CANRxBuffer.h"
+#include "abstract_task.h"
 
 void read_can_demo_task(void *pvParameter);
```
Comment (KarlSchu) [inline lib/CAN/CANBus.h:8]: remove it or is it used anywhere?

**Review by KarlSchu — APPROVED** [review]

File: lib/CAN/CANBus.cpp (hunk 1 above)
Comment (garned) [reply lib/CAN/CANBus.cpp:36]: Yes

**Pull request merged**